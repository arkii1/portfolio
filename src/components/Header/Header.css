.header {
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100vh;
  gap: 1rem;
}

.header__h1 {
  text-align: center;
  padding: 0 5rem;
  font-size: 2.2rem;
  word-wrap: break-word;
  color: var(--c-blue);
  font-weight: 700;
}

.header__secondary {
  text-align: center;
  padding: 0 5rem;
  font-size: 1.5rem;
  word-wrap: break-word;
  color: var(--c-white);
}

.header__button {
  color: var(--c-black);
  background-color: var(--c-blue);
  font-size: 1rem;
  border-radius: 10px;
  padding: 5px 15px;
  outline: none;
  border: solid 2px var(--c-blue);
  transition: 0.1s;
}

.header__button:hover {
  background-color: var(--c-white);
  color: var(--c-black);
  border: solid 2px var(--c-white);
  transform: scale(1.1);
  cursor: pointer;
}

.header__button:active {
  color: var(--c-white);
  border: solid 2px var(--c-white);
}