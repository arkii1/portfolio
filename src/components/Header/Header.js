import React from "react"
import "./Header.css"

function Header() {
  const onButtonClick = () => {
    document.querySelector(".about").scrollIntoView({ behavior: "smooth" })
  }

  return (
    <header className="header">
      <h1 className="header__h1">Tommy Mapp </h1>
      <p className="header__secondary">Full Stack Web Developer</p>
      <button className="header__button" type="button" onClick={onButtonClick}>
        Get to know me
      </button>
    </header>
  )
}

export default Header
