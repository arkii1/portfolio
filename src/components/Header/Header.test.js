import React from "react"
import ReactDOM from "react-dom/client"
import { cleanup } from "@testing-library/react"
import renderer from "react-test-renderer"
import Header from "./Header"

afterEach(cleanup)

it("renders without crashing", () => {
  const container = document.createElement("div")
  const root = ReactDOM.createRoot(container)
  root.render(<Header />)
})

it("matches snapshot", () => {
  const tree = renderer.create(<Header />).toJSON()
  expect(tree).toMatchSnapshot()
})
