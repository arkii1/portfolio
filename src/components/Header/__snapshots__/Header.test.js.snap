// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<header
  className="header"
>
  <h1
    className="header__h1"
  >
    Tommy Mapp 
  </h1>
  <p
    className="header__secondary"
  >
    Full Stack Web Developer
  </p>
  <button
    className="header__button"
    onClick={[Function]}
    type="button"
  >
    Get to know me
  </button>
</header>
`;
